fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraTransactionImpl.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl;
+
+import org.fcrepo.kernel.api.FedoraTransaction;
+
+/**
+ * The Fedora Transaction implementation
+ *
+ * @author mohideen
+ */
+public class FedoraTransactionImpl implements FedoraTransaction {
+
+    final String id;
+
+    FedoraTransactionImpl(final String id) {
```

Comment:
Perhaps assert non-null value here.

Reply:
@dbernstein Should it throw an Exception if the id is null or empty?

Reply:
I think it is best to throw an  IllegalArgumentException